tap_google_search_console/client.py

```diff
@@ -1,312 +1,107 @@
-            self.post(path=path, data=json.dumps(body))
+    def check_sites_access(self) -> None:
+        """Perform access check for each site url provided"""
+        body = json.dumps({"startDate": "2021-04-01", "endDate": "2021-05-01"})
+        for _ in self.__site_urls.replace(" ", "").split(","):
+            self.post(f"sites/{quote(_, safe='')}/searchAnalytics/query", data=body)
 
-    # during 'Timeout' error there is also possibility of 'ConnectionError',
-    # hence added backoff for 'ConnectionError' too.
-    @backoff.on_exception(backoff.expo,
-                          (Timeout, ConnectionError),
-                          max_tries=5,
-                          factor=2)
+    @backoff.on_exception(backoff.expo, Server5xxError, max_tries=5, factor=2)
     def __enter__(self):
         self.get_access_token()
         return self
 
     def __exit__(self, exception_type, exception_value, traceback):
         self.__session.close()
 
-    @backoff.on_exception(backoff.expo,
-                          Server5xxError,
-                          max_tries=5,
-                          factor=2)
-    def get_access_token(self):
-        # The refresh_token never expires and may be used many times to generate each access_token
-        # Since the refresh_token does not expire, it is not included in get access_token response
-        if self.__access_token is not None and self.__expires > datetime.utcnow():
-            return
-
-        headers = {}
-        if self.__user_agent:
-            headers['User-Agent'] = self.__user_agent
+    def get_access_token(self) -> None:
+        """Performs authentication and the access token if expired"""
 
+        if self.__access_token and self.__expires > datetime.now(timezone.utc):
+            return
+        headers = {"User-Agent": self.__user_agent or ""}
         response = self.__session.post(
             url=GOOGLE_TOKEN_URI,
             headers=headers,
             data={
-                'grant_type': 'refresh_token',
-                'client_id': self.__client_id,
-                'client_secret': self.__client_secret,
-                'refresh_token': self.__refresh_token,
+                "grant_type": "refresh_token",
+                "client_id": self.__client_id,
+                "client_secret": self.__client_secret,
+                "refresh_token": self.__refresh_token,
             },
-            timeout=self.request_timeout)
+            timeout=self.request_timeout,
+        )
 
         if response.status_code != 200:
             raise_for_error(response)
 
         data = response.json()
-        self.__access_token = data['access_token']
-        self.__expires = datetime.utcnow() + timedelta(seconds=data['expires_in'])
-        LOGGER.info('Authorized, token expires = {}'.format(self.__expires))
+        self.__access_token = data["access_token"]
+        self.__expires = datetime.now(timezone.utc) + timedelta(seconds=data["expires_in"])
```

Comment:
should use singer.utils.now instead of datetime.now()

Reply:
Done